core/lib/multivm/src/dump.rs

```diff
@@ -0,0 +1,160 @@
+use std::{collections::HashMap, sync::Arc};
+
+use serde::{Deserialize, Serialize};
+use zksync_types::{web3, L1BatchNumber, Transaction, H256, U256};
+use zksync_utils::u256_to_h256;
+use zksync_vm_interface::{
+    storage::{InMemoryStorage, ReadStorage, StoragePtr, StorageView},
+    L1BatchEnv, L2BlockEnv, SystemEnv, VmFactory,
+};
+
+/// Handler for [`VmDump`].
+pub type VmDumpHandler = Arc<dyn Fn(VmDump) + Send + Sync>;
```

Comment:
You may introduce a wrapper type that would implement `Debug` to not force any type that contains it to have a custom `Debug` implementation.